src/java.base/share/classes/java/lang/invoke/VarForm.java

```diff
@@ -93,14 +109,26 @@ final MethodType getMethodType(int type) {
 
     @ForceInline
     final MemberName getMemberName(int mode) {
-        // TODO calculate lazily
         MemberName mn = memberName_table[mode];
         if (mn == null) {
-            throw new UnsupportedOperationException();
+            mn = resolveMemberName(mode);
         }
         return mn;
     }
 
+    @DontInline
+    private final MemberName resolveMemberName(int mode) {
+        AccessMode value = AccessMode.values()[mode];
+        String methodName = value.methodName();
+        try {
+            MethodType type = methodType_table[value.at.ordinal()].insertParameterTypes(0, VarHandle.class);
+            return memberName_table[mode]
+                = MethodHandles.Lookup.IMPL_LOOKUP
+                    .resolveOrFail(REF_invokeStatic, implClass, methodName, type);
+        } catch (NoSuchMethodException | IllegalAccessException e) {
+            throw new UnsupportedOperationException();
+        }
```

Comment:
```suggestion
        } catch (ReflectiveOperationException e) {
                throw newInternalError("Failed resolving VarHandle member name", ex);
        }
```

Reply:
Thanks for reviewing!

Is there's a way to provoke this exception through the public API? If not then the suggested behavior change seems reasonable.

Reply:
No, since VarHandles are not publicly extensible, the exception should not occur unless something has gone very wrong (the correspondence between access mode and implementing method is broken).

Reply:
Unfortunately the change to InternalError breaks a number of tests, since the UOE does bubble up through the public API. I also found a few failing tests I had overlooked due VarHandle.isAccessModeSupported throwing rather than returning false, so I had to slightly rework the patch.

Reply:
Oops, my bad. I got confused and forgot that VH implementations can avoid implementing access mode methods that would just throw UOE. This slightly complicates lazily resolution.

We don't cache the result of a failed method resolution, which would require a non-null sentinel value, probably does not matter.

Using `resolveOrNull` seems a better fit rather than catching and dropping ROE, less work performed too.